src/compas_rhino/artists/artist.py

```diff
@@ -17,7 +18,10 @@
 __all__ = ['Artist']
 
 
-class Artist(object):
+_ITEM_ARTIST = {}
+
+
+class Artist(Object3D):
```

Comment:
This feels very confusing. Why is `Artist` inheriting from `Object3D`? Maybe is correct, but I just find it rather unnatural...

Reply:
shouldn't keeping track of parent-child relationships, transformation stack, ... be a general responsibility of COMPAS objects rather than of the visualisation objects only?

Reply:
It is a bit... Another way is to have Object3D under an artist its attribute, but that would create a double layer wrapper situation when adding a geometry directly under an artist as children:  it has to firstly wrap a Object3D around geometry, then a child artist around Object3D, and then add it to the parent artist.  At same time, to access local geometry info from artist would also become quite deep like `artist.object3d.line.start.x`

Reply:
ideally you don't go through that hierarchy at all (see the new plotters). you just update the geometry directly and this will affect the drawing in the next drawing cycle. the artist only provides access to all attributes related to representation...

Reply:
> you just update the geometry directly and this will affect the drawing in the next drawing cycle

makes more sense to me

Reply:
> makes no sense to me

as in "this is a really bad idea!", or "how would that work?!"

Reply:
hehe.. it's the other way around, it's *more*, not *no*! ;)  It does make sense to me

Reply:
ha sorry, misread :)

Reply:
i think we should first look at a few usage scenarios in some (dummy) snippets across the various interfaces rather than immediately dive in with the implementations like this.

perhaps a few gists to compare?

Reply:
haha agree, i m writing up some

Reply:
👍  API-first design! ;)

Reply:
@gonzalocasas @brgcode 
https://gist.github.com/Licini/3c8ca96a68705414b00f81f06b458fe2

Reply:
After writing few snippets, starting to agree having artists inheriting object3d was indeed too weird. Actually artist as attribute of object3d makes more sense to me at this point. Also I agree a lot people should never access geometry through an artist which is basically a utility only for visualisation purpose.

Reply:
I'm just reposting those snippets here again from slack in case anyone cannot find them:

example of auto-generating artist from object3d: https://gist.github.com/Licini/ddaba6afe47b0b56d09d118399347d21
some ideas about “Collection” class and “Group” class
https://gist.github.com/Licini/f9d8a6aa7bc9d906d1e9a066e2935a84
some possible simplified drawing apis:
https://gist.github.com/Licini/f7b0c3a5c9e39db850ae18f0d0d42fdb
create and update object3ds from CAD:
https://gist.github.com/Licini/258f3a6e655ab27ccca05ddceaee0752